### DISCUSSION TITLE
Add Pdf template and attachment link to rows

### DISCUSSION THREAD

**Discussion by rmutili** [Budibase/budibase#1948] [Ideas]
Is it possible to add pdf attachment link to a row. This data is coming from a form filled and it generates a pdf document based on a template with variables. The pdf links could be based on approval workflow where each column can have specific links for different approvals(Manager Approval, Finance Approval, etc)

**Comment by joebudi** [comment]
Hey @rmutili 

Thank you for creating a discussion. Could you clarify a few things:

How are you generating the PDFs (where are the templates/variables)? 
What database are you using with Budibase?

Cheers
Joe

**Reply by rmutili** [reply]
Hey @joebudi , thanks for jumping in.

The PDFs need to be generated from a workflow. A user fills in a form, goes to manager for approval. The manager receives a notification with a pdf attached of the content of the filled form. 

The templates need to be defined as the form which will pick variables from the table.

**Reply by joebudi** [reply]
Thanks for the info. Currently, there is no way to generate a PDF with Budibase right now.

An alternative approach would be to:

-  create an additional column in the table, called approval
(if you have generated a form from the table, you will need to remove the approval field from the form (which users fill out)).
- create a list screen and a detail screen (with access filter set to admin)
- add the manager as a user with an admin role.

This way, the manager can sign off the form submission and there's no need for a PDF.

I'm not sure if this solution suffices but I thought I'd offer it as it might be useful.

Joe

**Reply by rmutili** [reply]
Thanks Joe,

Yes this approach may work. I will be interested if you could reproduce the same and share some screenshots.

Is there a way to have three approval options ("Reject"(With a reason), "Need Correction"(With a description) and "Approved".

1. Reject - Request is discarded
2. Need Correction - Requester can amend the request
3. Approved - Moves to the next approval stage

**Reply by joebudi** [reply]
Hey @rmutili 

Apologies for the late reply. Here is a quick video that may help:
https://www.loom.com/share/4e9184f57d3e4a19aff3cfa24d558a0c

**Reply by rmutili** [reply]
Thanks @joebudi, let me reproduce it and give u feedback... Otherwise thanks for jumping in...

**Comment by rmutili** [comment]
Hey @joebudi, came across this https://apitemplate.io/... Not sure if it can be integrated to generate a pdf after a column is edited. Push the content to apitemplate, which generates a pdf and sends back the link to the url which can be updated on the row... Sounds like a work around...

**Reply by Bogey** [reply]
Or an easy way to push it to docassemble (like https://docassemble.org/docs/api.html#session_post)

**Reply by rmutili** [reply]
Sounds like another idea... Thanks @Bogey

**Reply by Bogey** [reply]
docassemble has useful features and I’ve been looking into a way to incorporate it into my workflow.

**Reply by rmutili** [reply]
@Bogey let me know how it goes..

**Reply by oswaldB** [reply]
@rmutili if you want to stay open source you can check https://carbone.io/
